### PR TITLE
[FrameworkBundle] Add missing resource tracking for validation mapping

### PR DESCRIPTION
**Pull request by chalasr**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | https://github.com/symfony/symfony/pull/19086#discussion_r104890636
| License       | MIT
| Doc PR        | n/a

https://github.com/symfony/symfony/pull/21927/files?w=1

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php [hunk 1]
```diff
@@ -1011,17 +1011,19 @@ private function getValidatorMappingFilesFromDir($dir, array &$files)
...
         foreach ($config['mapping']['paths'] as $path) {
-            if (is_dir($path)) {
-                $this->getValidatorMappingFilesFromDir($path, $files);
-            } elseif (is_file($path)) {
-                if (preg_match('/\.(xml|ya?ml)$/', $path, $matches)) {
-                    $extension = $matches[1];
-                    $files['yaml' === $extension ? 'yml' : $extension][] = $path;
-                } else {
-                    throw new \RuntimeException(sprintf('Unsupported mapping type in "%s", supported types are XML & Yaml.', $path));
+            if ($container->fileExists($path)) {
```
Comment (stof) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php]: content needs to be tracked for directories (but only for new XML and YAML files). But it does not need to be tracked for files (changing the content of the validation file does not need to invalidate the container). So this shoould be improved.

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php [hunk 2]
```diff
@@ -1011,17 +1011,19 @@ private function getValidatorMappingFilesFromDir($dir, array &$files)
...
+                if (is_dir($path)) {
+                    $this->getValidatorMappingFilesFromDir($path, $files);
+                } elseif (is_file($path)) {
```
Comment (stof) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php]: ``else`` is enough here

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php (hunk 1 above)
Comment (chalasr) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php]: updated. Could you please have a look?

**Review by nicolas-grekas — APPROVED** [review]
:+1:

**Comment by nicolas-grekas** [comment]
Thank you @chalasr.

**Pull request merged**